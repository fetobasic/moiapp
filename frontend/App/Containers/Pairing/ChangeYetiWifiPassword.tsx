import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

import { Button, HeaderSimple as Header, InfoRow, Pressable, TextInputPassword } from 'App/Components';

import { ApplicationStyles, Colors, Fonts, Metrics } from 'App/Themes';
import { PairingStackParamList } from 'App/Types/NavigationStackParamList';
import renderElement from 'App/Services/RenderElement';

import WiFiIcon from 'App/Images/Icons/wifi.svg';
import { useAppDispatch, useAppSelector, usePairingNavigation, usePrevious } from 'App/Hooks';
import { yetiActions } from 'App/Store/Yeti';

type Props = NativeStackScreenProps<PairingStackParamList, 'ChangeYetiWifiPassword'>;

function ChangeYetiWifiPassword({ route }: Props) {
  const dispatch = useAppDispatch();
  const navigation = usePairingNavigation('ChangeYetiWifiPassword');
  const { inProgress, error } = useAppSelector((state) => ({
    inProgress: state.yetiInfo.passwordChange.inProgress,
    error: state.yetiInfo.passwordChange.error,
  }));

  const [password, setPassword] = useState('');
  const prevInProgress = usePrevious(inProgress);

  const isEmptyPassword = useMemo(() => password === '', [password]);

  const handleNextButton = useCallback(() => {
    if (isEmptyPassword) {
      navigation.navigate('SelectWifiNetwork', {
        ...route.params,
      });
    } else {
      dispatch(yetiActions.yetiSetPassword.request({ password }));
    }
  }, [dispatch, isEmptyPassword, navigation, password, route.params]);

  useEffect(() => {
    if (prevInProgress && !inProgress && !error) {
      navigation.navigate('SelectWifiNetwork', {
        ...route.params,
      });
    }
  }, [error, inProgress, navigation, prevInProgress, route.params]);

  return (
    <SafeAreaView edges={['bottom']} style={ApplicationStyles.mainContainer}>
      <Header
        title="Wi-Fi"
        rightSection={renderElement(
          <Pressable onPress={handleNextButton}>
            <Text style={styles.textHeaderRight}>{isEmptyPassword ? 'Skip' : 'Save'}</Text>
          </Pressable>,
        )}
      />
      <View style={ApplicationStyles.container}>
        <View style={ApplicationStyles.flex}>
          <InfoRow
            withBorder={false}
            icon={renderElement(<WiFiIcon />)}
            title="Change Password"
            body={renderElement(
              <Text style={styles.textBody}>
                You have already joined the Yeti's network. This step allows you to change the default{' '}
                <Text style={styles.textHighlightGreen}>GOALZERO</Text> to your own secure password.{'\n\n'}
                <Text style={styles.textHighlightYellow}>Note</Text>: Password must be at least 8 characters long
                consisting of numbers and letters.
              </Text>,
            )}
          />
          <TextInputPassword
            focused
            onSubmitEditing={handleNextButton}
            placeholder="Password"
            value={password}
            onChangeText={setPassword}
            returnKeyType="go"
          />
        </View>
        <Button title="Next" onPress={handleNextButton} />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  textBody: {
    marginTop: Metrics.smallMargin,
    ...Fonts.font.base.bodyOne,
    color: Colors.transparentWhite('0.6'),
  },
  textHighlightGreen: {
    color: Colors.green,
  },
  textHighlightYellow: {
    color: Colors.severity.yellow,
  },
  textHeaderRight: {
    ...Fonts.font.base.subtitleOne,
    color: Colors.green,
  },
});

export default ChangeYetiWifiPassword;
