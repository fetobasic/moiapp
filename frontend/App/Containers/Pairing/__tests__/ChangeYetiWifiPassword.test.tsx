import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import ChangeYetiWifiPassword from '../ChangeYetiWifiPassword';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('ChangeYetiWifiPassword', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
      },
      route: { params: {} },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <ChangeYetiWifiPassword {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
