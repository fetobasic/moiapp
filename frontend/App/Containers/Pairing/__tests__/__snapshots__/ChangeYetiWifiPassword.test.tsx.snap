// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeYetiWifiPassword renders screen 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "off",
      "right": "off",
      "top": "off",
    }
  }
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Wi-Fi
        </Text>
      </View>
      <View
        style={
          {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "opacity": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "rgba(191, 210, 43, 1)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              }
            }
          >
            Skip
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgba(27, 27, 27, 1)",
        "flex": 1,
        "paddingBottom": 0,
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgba(27, 27, 27, 1)",
              "borderRadius": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 15,
            },
            undefined,
            {
              "marginBottom": 8,
              "marginHorizontal": 1,
              "paddingVertical": 16,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 32,
                "justifyContent": "center",
                "width": 32,
              }
            }
          >
            <SvgMock />
          </View>
          <View
            style={
              [
                {
                  "flex": 1,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                undefined,
                {
                  "paddingRight": 0,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginVertical": 4,
                  },
                  undefined,
                ]
              }
            >
              Change Password
            </Text>
            <Text
              style={
                {
                  "color": "rgba(255, 255, 255, 0.6)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginTop": 8,
                }
              }
            >
              You have already joined the Yeti's network. This step allows you to change the default
               
              <Text
                style={
                  {
                    "color": "rgba(191, 210, 43, 1)",
                  }
                }
              >
                GOALZERO
              </Text>
               to your own secure password.
              


              <Text
                style={
                  {
                    "color": "rgba(255, 205, 29, 1)",
                  }
                }
              >
                Note
              </Text>
              : Password must be at least 8 characters long consisting of numbers and letters.
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(27, 27, 27, 1)",
              "elevation": 12,
              "height": 56,
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 12,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "borderColor": "rgba(54, 54, 54, 1)",
                "borderRadius": 8,
                "borderWidth": 1,
                "flex": 1,
                "height": 56,
                "justifyContent": "center",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                {
                  "backgroundColor": "rgba(27, 27, 27, 1)",
                  "color": "rgba(255, 255, 255, 0.871)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginLeft": 8,
                  "paddingHorizontal": 4,
                  "position": "absolute",
                  "top": 16,
                }
              }
            >
              Password
            </Text>
            <TextInput
              accessibilityLabel="Password field"
              accessible={true}
              autoFocus={true}
              focused={true}
              keyboardAppearance="dark"
              maxFontSizeMultiplier={1.15}
              maxLength={200}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onPressIn={[Function]}
              onSubmitEditing={[Function]}
              placeholder=""
              placeholderTextColor="rgba(255, 255, 255, 0.38)"
              returnKeyType="go"
              secureTextEntry={true}
              selectionColor="rgba(191, 210, 43, 1)"
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 16,
                    "paddingHorizontal": 12,
                  },
                  [
                    {
                      "paddingRight": 40,
                    },
                    undefined,
                  ],
                ]
              }
              testID="passwordInput"
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
          </View>
        </View>
        <View
          accessibilityLabel="Eye watermark"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "opacity": 1,
              },
              [
                {
                  "bottom": 5,
                  "padding": 10,
                  "position": "absolute",
                  "right": 0,
                },
                undefined,
              ],
            ]
          }
        >
          <SvgMock
            color="rgba(255,255,255, 0.87)"
            testID="eyeIcon"
          />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "opacity": 1,
          },
          [
            {
              "elevation": 12,
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 12,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          {
            "borderRadius": 24,
            "overflow": "hidden",
          }
        }
      >
        <RNSVGSvgView
          bbHeight="48"
          bbWidth="100%"
          focusable={false}
          height={48}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 48,
                "width": "100%",
              },
            ]
          }
          width="100%"
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0,
                    -13224394,
                    1,
                    1776411,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="grad"
                x1="0"
                x2="0"
                y1="0"
                y2="0.5"
              />
            </RNSVGDefs>
            <RNSVGRect
              fill={
                {
                  "brushRef": "grad",
                  "type": 1,
                }
              }
              height="48"
              propList={
                [
                  "fill",
                ]
              }
              width="100%"
              x="0"
              y="0"
            />
            <View
              style={
                [
                  {
                    "borderRadius": 24,
                    "marginHorizontal": 1,
                    "marginTop": 1,
                  },
                  {
                    "backgroundColor": "rgba(191, 210, 43, 1)",
                    "height": 46,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingVertical": 12,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(255, 255, 255, 1)",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 14,
                          "fontWeight": "700",
                          "lineHeight": 20,
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "rgba(27, 27, 27, 1)",
                        },
                        undefined,
                      ]
                    }
                  >
                    Next
                  </Text>
                </View>
              </View>
            </View>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
